NBitcoin.Altcoins/ForkIdTransaction.cs

```diff
@@ -47,7 +47,7 @@ public override uint256 GetSignatureHash(Script scriptCode, int nIn, SigHash nHa
 			if(UsesForkId(nHashType))
 				nForkHashType |= ForkId << 8;
 
-			if((SupportSegwit && sigversion == HashVersion.Witness) || UsesForkId(nHashType))
+			if((SupportSegwit && sigversion == HashVersion.Witness) || (UsesForkId(nHashType) && ForkId != 42))
```

Comment:
As I said, do not do this. Make your own class which derive from ForkId, and overrride `UsesFork` to always returns true.